### PR TITLE
Disable no-undef

### PR DESCRIPTION
**Pull request by nolanleung**
It is recommended to disable `no-undef` in TypeScript projects, as documented here:

[We strongly recommend that you do not use the no-undef lint rule on TypeScript projects. The checks it provides are already provided by TypeScript without the need for configuration - TypeScript just does this significantly better.](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors)

### CODE REVIEW

**Comment by kaimallea** [comment]
@nolanleung I know we're tracking this internally, but can you add a description for public eyes? 🙏🏾

**Review by kaimallea — APPROVED** [review]

**Comment by kaimallea** [comment]
Just realized I didn't review the commit message formatting 😞  since it doesn't follow the [convention](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines), semantic-release won't trigger a publish. 

tl;dr should've used `yarn commit` to trigger the commitizen prompts
